Xamarin.PropertyEditing.Mac/Controls/Custom/PropertyButton.cs

```diff
@@ -236,5 +255,19 @@ private void OnResourceRequested (object sender, EventArgs e)
 
 			resourceSelectorPopOver.Show (requestResourceView.Frame, (NSView)this, NSRectEdge.MinYEdge);
 		}
+
+		private void OnBindingRequested (object sender, CreateBindingRequestedEventArgs e)
+		{
+			var bindingEditorWindow = new BindingEditorWindow (this.hostResources, this.viewModel) {
+				Appearance = EffectiveAppearance,
+			};
+
+			var result = (NSModalResponse)(int)NSApplication.SharedApplication.RunModalForWindow (bindingEditorWindow);
+			if (result == NSModalResponse.OK) {
+				e.BindingObject = bindingEditorWindow.ViewModel?.SelectedObjects?.Single ();
```

Comment:
Neither the view model nor `SelectedObjects` should ever be null, so why are we treating them like they might be?

Reply:
Just being defensive.

Reply:
To me being defensive means things continue to work in an unforeseen circumstance. This won't do that, the error already occurred by these being null, and so it'll just hide the exception making it harder to find.

Reply:
Ok.